src/main/resources/assets/computercraft/lua/rom/apis/window

```diff
@@ -268,16 +268,19 @@ function create( parent, nX, nY, nWidth, nHeight, bStartVisible )
     function window.isColour()
         return isColor()
     end
-
+ 
+    local floor, log, log2 = math.floor, math.log, math.log(2)
     local function setTextColor( color )
-        if type(color) == "number" then
-	    nTextColor = color
-            if bVisible then
-               updateCursorColor()
-            end
-        else
-           error("Color must be a number")
-        end
+	if not(type(color) == "number") then
```

Comment:
Also, not entirely sure what is going on with the indentation here - it seems like a really weird mix of spaces and tabs.